torch/csrc/jit/codegen/cuda/transform_replay.h

```diff
@@ -153,36 +155,92 @@ class TORCH_CUDA_CU_API TransformReplay {
       const TensorDomain* self);
 };
 
+// TransformPropagator starts from a reference tensor, and propagate
+// the transformations in this tensor to the entire graph. The propagation
+// is done with the Dijkstra algorithm which will transform every tensor
+// in this graph based on the information flow from the path that perserves
+// the most amount of information about the reference tensor. Every tensor in
+// the graph is replayed only once.
+//
+// During the propagation, we explicitly keep track of the information about
+// which reference tensor's root ID's information is preserved, and to which
+// level. This information is stored as a vector of `RootIDInfo`, where each
+// item in the vector correspond to one ID in the reference tensor's root
+// domain.
 class TORCH_CUDA_CU_API TransformPropagator {
- private:
-  bool replayPasC(TensorView* producer_tv, TensorView* consumer_tv = nullptr);
-  bool replayCasP(TensorView* consumer_tv, TensorView* producer_tv = nullptr);
+ public:
+  // This is a struct storing how the information about a root ID in the
+  // starting tensor is preserved during propagation. If during propagation, we
+  // reached a tensor called the "current" tensor, we are interested in the
+  // following information:
+  // - Which reference tensor's root ID's information does the current tensor
+  //   contains?
+  // - For each of these root IDs above, what are its corresponding IDs in the
+  //   current tensor's root/rfactor domain?
+  // - Is the current tensor's information about this reference tensor's root ID
+  //   complete?
+  struct RootIDInfo {
+    // An ID in the root domain of the reference tensor
+    IterDomain* starting_root_id;
```

Comment:
This is actually not needed. And I just removed it. This is conceptually helpful because it tells us that each object of `RootIDInfo` corresponds to one root ID in the reference tensor, but storing the pointer for that ID is actually not necessary. I updated the comment accordingly.